wlog/auditlog/audit3log/params.go

```diff
@@ -0,0 +1,216 @@
+		entry.OptionalStringValue(ProducerTypeKey, string(producerType))
+	})
+}
+
+func Organizations(organizations ...AuditOrganizationType) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(OrganizationsKey, organizations, reflect.TypeOf(organizations))
+	})
+}
+
+func EventID(eventID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(EventIDKey, eventID)
+	})
+}
+
+func UserAgent(userAgent string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(UserAgentKey, userAgent)
+	})
+}
+
+func Categories(categories ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(CategoriesKey, categories)
+	})
+}
+
+func Entities(entities ...interface{}) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(EntitiesKey, entities, reflect.TypeOf(entities))
+	})
+}
+
+func Users(users ...AuditContextualizedUserType) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(UsersKey, users, reflect.TypeOf(users))
+	})
+}
+
+func Origins(origins ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(OriginsKey, origins)
+	})
+}
+
+func SourceOrigin(sourceOrigin string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(SourceOriginKey, sourceOrigin)
+	})
+}
+
+func UID(uid string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.UIDKey, uid)
+	})
+}
+
+func SID(sid string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.SIDKey, sid)
+	})
+}
+
+func TokenID(tokenID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.TokenIDKey, tokenID)
+	})
+}
+
+func TraceID(traceID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.TraceIDKey, traceID)
+	})
+}
+
+func OtherUIDs(otherUIDs ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(OtherUIDsKey, otherUIDs)
+	})
+}
+
+func Origin(origin string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(OriginKey, origin)
+	})
+}
+
+func RequestParam(key string, levels []AuditSensitivityType, payload interface{}) Param {
+	return RequestParams(map[string]AuditSensitivityTaggedValueType{
+		key: {levels, payload},
+	})
+}
+
+func RequestParams(requestParams map[string]AuditSensitivityTaggedValueType) Param {
+	requestParamsInterface := make(map[string]interface{})
+	for key, val := range requestParams {
+		requestParamsInterface[key] = val
+	}
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.AnyMapValue(RequestParamsKey, requestParamsInterface)
+	})
+}
+
+func ResultParam(key string, levels []AuditSensitivityType, payload interface{}) Param {
+	return ResultParams(map[string]AuditSensitivityTaggedValueType{
```

Comment:
Calling into ResultParams is syntactically clean but results in an unnecessary map allocation. Should this just be 
```go 
	return paramFunc(func(entry wlog.LogEntry) {
		entry.AnyMapValue(ResultParamsKey, map[string]interface{}{key: AuditSensitivityTaggedValueType{levels, payload}})
	})
```
(same q for RequestParam above)

Reply:
Done!